infra-gen2/tool/deploy_gen2.dart

```diff
@@ -0,0 +1,391 @@
+#!/usr/bin/env dart
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:path/path.dart' as p;
+
+class AmplifyIntegCategory {
+  const AmplifyIntegCategory({
+    required this.category,
+    required this.backends,
+    required this.defaultOutput,
+    this.sharedOutputs = const [],
+  });
+
+  /// This is the category of the integration group
+  final Category category;
+
+  /// This is the list of backends for the category
+  final List<AmplifyBackend> backends;
+
+  /// This is the default output path for the category
+  final String defaultOutput;
+
+  /// This is the list of shared output paths for the category
+  final List<String> sharedOutputs;
+}
+
+class AmplifyBackend {
+  const AmplifyBackend({
+    required this.name,
+    required this.identifier,
+    required this.pathToSource,
+  });
+
+  /// This is the name of the backend
+  final String name;
+
+  /// This is the identifier for the backend
+  /// It is the name in "PascalCase"
+  /// Must be less than 15 characters
+  final String identifier;
+
+  /// This is the path to the source code for the backend
+  /// Root directory is amplify-flutter/
+  final String pathToSource;
+}
+
+/// This is the source of truth for the infra-gen2 backends
+///
+/// To add a backend:
+/// 1. Create the backend in the infra-gen2/backends directory
+/// 2. Run `npm create amplify@latest -y` in the backend directory to create the backend
+/// 3. Add the backend to a category or create a new category
+/// 4. Run `dart tool/deploy_gen2.dart` to deploy the backend
+const List<AmplifyIntegCategory> infraConfig = [
+  AmplifyIntegCategory(
+    category: Category.analytics,
+    backends: [],
+    defaultOutput: '',
+  ),
+  AmplifyIntegCategory(
+    category: Category.api,
+    backends: [
```

Comment:
Q: Which backend becomes the "default" (the one set to `const amplifyConfig = ... ` in the outputs file? I added a new backend and that one is now the "default". Maybe it would be beneficial to specify one as default?

Reply:
It's currently implicit, as the first one in the list will be default. I'm not sure its work adding more logic to explicitly set the default since this script is typically used once-per team member.